### PR TITLE
DSS Outcomes repository review, refactor and cleanup

### PR DESCRIPTION
**Pull request by mimran-DFE**
DSS Outcomes repository review, refactor and cleanup

### CODE REVIEW

File: NCS.DSS.Outcomes/Cosmos/Provider/CosmosDBProvider.cs
Comment (hnandra-DFE) [inline NCS.DSS.Outcomes/Cosmos/Provider/CosmosDBProvider.cs:1]: Requires logging similar to dss-learningprogressions: https://github.com/SkillsFundingAgency/dss-learningprogression/pull/24/files#diff-fa5bdeeb484b0b4b7e9a745b75067149baf0cff532f4ddfedd02fb09dbee1cf5

File: NCS.DSS.Outcomes/GetOutcomesByIdHttpTrigger/Function/GetOutcomesByIdHttpTrigger.cs [hunk 1]
```diff
@@ -137,7 +137,9 @@ public async Task<IActionResult> RunAsync([HttpTrigger(AuthorizationLevel.Anonym
             _logger.LogInformation($"Function {nameof(GetOutcomesByIdHttpTrigger)} has finished invocation");
 
             if (outcomes == null)
+            {
                 return new NoContentResult();
+            }
 
             return new JsonResult(outcomes, new JsonSerializerOptions())
             {
```
Comment (hnandra-DFE) [inline NCS.DSS.Outcomes/GetOutcomesByIdHttpTrigger/Function/GetOutcomesByIdHttpTrigger.cs:145]: Need to return logging of retrieving the outcome is successful with its outcome ID

File: NCS.DSS.Outcomes/GetOutcomesHttpTrigger/Function/GetOutcomesHttpTrigger.cs
Comment (hnandra-DFE) [inline NCS.DSS.Outcomes/GetOutcomesHttpTrigger/Function/GetOutcomesHttpTrigger.cs:1]: Need to return logging of retrieving the outcomes is successful with outcome IDs

File: NCS.DSS.Outcomes/PatchOutcomesHttpTrigger/Function/PatchOutcomesHttpTrigger.cs
Comment (hnandra-DFE) [inline NCS.DSS.Outcomes/PatchOutcomesHttpTrigger/Function/PatchOutcomesHttpTrigger.cs:1]: When a 4XX response is returned, the log level should be warning for the log
When an exception is returned, the log should return the exception object and the exception message
Needs entry and exit logging on reading the request L134-L143
Needs successful and unsuccessful on the validation
Needs a log to return that the PATCH was unsuccessful

File: NCS.DSS.Outcomes/PostOutcomesHttpTrigger/Function/PostOutcomesHttpTrigger.cs
Comment (hnandra-DFE) [inline NCS.DSS.Outcomes/PostOutcomesHttpTrigger/Function/PostOutcomesHttpTrigger.cs:1]: When a 4XX response is returned, the log level should be warning for the log
When an exception is returned, the log should return the exception object and the exception message
Needs a successful log when reading the request body
Needs successful and unsuccessful on the validation
The log to log success is before the service bus post so it is invalid and must be at the end of the return of the response

**Review by hnandra-DFE — CHANGES_REQUESTED** [review]

**Comment by hnandra-DFE** [comment]
You can follow the logging I have applied in dss-learningprogressions here: https://github.com/SkillsFundingAgency/dss-learningprogression/pull/24

**Review by ajones-DFE — APPROVED** [review]

**Pull request merged**